src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.ref.cs

```diff
@@ -60,7 +60,7 @@ public static partial class CacheEntryExtensions
     public static partial class CacheExtensions
     {
         public static object? Get(this Microsoft.Extensions.Caching.Memory.IMemoryCache cache, object key) { throw null; }
-        public static System.Threading.Tasks.Task<TItem> GetOrCreateAsync<TItem>(this Microsoft.Extensions.Caching.Memory.IMemoryCache cache, object key, System.Func<Microsoft.Extensions.Caching.Memory.ICacheEntry, System.Threading.Tasks.Task<TItem>> factory) { throw null; }
+        public static System.Threading.Tasks.Task<TItem?> GetOrCreateAsync<TItem>(this Microsoft.Extensions.Caching.Memory.IMemoryCache cache, object key, System.Func<Microsoft.Extensions.Caching.Memory.ICacheEntry, System.Threading.Tasks.Task<TItem>> factory) { throw null; }
```

Comment:
What about this nullability difference? The reference source didn't mark the generic type as nullable but the source does. @maxkoshevoi @eerhardt

Reply:
Thanks! Updated [PR](https://github.com/dotnet/runtime/pull/65804)

Reply:
Hm, thinking about it

```cs
TItem Foo(Func<ICacheEntry, TItem> factory)
{
  var result = factory(entry)
  return result; // Why does VS warn that result may be null here?
}
```

Maybe `GetOrCreate` and `GetOrCreateAsync` cannot return null after all (if TItem is non-nullable, and can if it is)?